### PR TITLE
Long string switch rebase help

### PR DESCRIPTION
**Pull request by lock9**
Hi @lightszero  @shargon,  I fixed the commits, can you check if this is correct? 
Original PR: #116 
Author: @lightszero

### CODE REVIEW

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs [hunk 1]
```diff
@@ -961,6 +954,151 @@ private int _ConvertCall(OpCode src, NeoMethod to)
...
+                        || code2.code == CodeEx.Ldc_I4_4 || code2.code == CodeEx.Ldc_I4_5 || code2.code == CodeEx.Ldc_I4_6 || code2.code == CodeEx.Ldc_I4_7 || code2.code == CodeEx.Ldc_I4_8
+                        || code2.code == CodeEx.Ldc_I4_M1 || code2.code == CodeEx.Ldc_I4_S;
+                    var bJmp = code3.code == CodeEx.Beq || code3.code == CodeEx.Beq_S || code3.code == CodeEx.Bge || code3.code == CodeEx.Bge_S || code3.code == CodeEx.Bge_Un || code3.code == CodeEx.Bge_Un_S
+                        || code3.code == CodeEx.Bgt || code3.code == CodeEx.Bgt_S || code3.code == CodeEx.Bgt_Un || code3.code == CodeEx.Bgt_Un_S
+                        || code3.code == CodeEx.Ble || code3.code == CodeEx.Ble_S || code3.code == CodeEx.Ble_Un || code3.code == CodeEx.Ble_Un_S
+                        || code3.code == CodeEx.Blt || code3.code == CodeEx.Blt_S || code3.code == CodeEx.Blt_Un || code3.code == CodeEx.Blt_Un_S
+                        || code3.code == CodeEx.Bne_Un || code3.code == CodeEx.Bne_Un_S;
```
Comment (lock9) [inline src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: I think that we could encapsulate this kind of logic, with something like "Is_ctor()".

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs [hunk 2]
```diff
@@ -961,6 +954,151 @@ private int _ConvertCall(OpCode src, NeoMethod to)
...
+                        skipcount += 3;
+                        jumptablecount++;
+                    }
+                    else
+                    {
+                        isjumptable = false;
+                        //順便看看是不是ldstr 段
+                        if (bLdLoc1 && code1.tokenI32 == last.tokenI32 && code2.code == CodeEx.Ldstr && (code3.code == CodeEx.Call || code3.code == CodeEx.Callvirt) && code3.tokenMethod.Is_String_op_Equality())
+                        {
+                            //is switch ldstr with ldloc
```
Comment (lock9) [inline src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: Is this supposed to be empty?

File: tests/Neo.Compiler.MSIL.UnitTests/Utils/NeonTestTool.cs [hunk 3]
```diff
@@ -43,13 +46,33 @@ public static byte[] HexString2Bytes(string str)
...
+            var comp = CSharpCompilation.Create("TestContract", new[] { tree }, new[]
             {
+                MetadataReference.CreateFromFile(Path.Combine(coreDir, "mscorlib.dll")),
+                MetadataReference.CreateFromFile(Path.Combine(coreDir, "System.Runtime.dll")),
+                MetadataReference.CreateFromFile(Path.Combine(coreDir, "System.Runtime.Numerics.dll")),
+                MetadataReference.CreateFromFile(typeof(System.ComponentModel.DisplayNameAttribute).Assembly.Location),
+
+                MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(OpCodeAttribute).Assembly.Location)
+           }, op);
```
Comment (lock9) [inline tests/Neo.Compiler.MSIL.UnitTests/Utils/NeonTestTool.cs]: Is this correct? Should it load the Neo Framework dll here too?

Comment (shargon) [reply tests/Neo.Compiler.MSIL.UnitTests/Utils/NeonTestTool.cs]: This should reuse the logic from `Compiler` class

Comment (shargon) [reply tests/Neo.Compiler.MSIL.UnitTests/Utils/NeonTestTool.cs]: @lightszero @lock9 If we want to allow `Debug` mode (I don't know why) we should change the `Compiler` class

Comment (lightszero) [reply tests/Neo.Compiler.MSIL.UnitTests/Utils/NeonTestTool.cs]: unittest need test both dll debug and release。

Comment (shargon) [reply tests/Neo.Compiler.MSIL.UnitTests/Utils/NeonTestTool.cs]: @lock9 https://github.com/simplitech/neo-devpack-dotnet/pull/2

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs [hunk 4]
```diff
@@ -961,6 +954,151 @@ private int _ConvertCall(OpCode src, NeoMethod to)
             }
             return 0;
         }
+
+        private int _ConvertStringSwitch(ILMethod method, OpCode src, NeoMethod to)
+        {
```
Comment (shargon) [inline src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: @lightszero could you add a comment with the expected IL code?

Comment (lightszero) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: I will do that later,maybe we can merge this first.

Comment (shargon) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: Don't forget it please 🙏😬

**Pull request merged**